pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -199,6 +199,13 @@ type CiliumBGPVirtualRouter struct {
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default=false
 	ExportPodCIDR *bool `json:"exportPodCIDR,omitempty"`
+	// PodIPPools is a list of IPAM pools to advertise that have been allocated to the
+	// CiliumNodes matching this policy. All CIDRs for a specified pool will be advertised.
+	//
+	// If unspecified, no IPAM pool CIDRs will be advertised.
+	//
+	// +kubebuilder:validation:Optional
+	PodIPPools []string `json:"podIPPools,omitempty"`
```

Comment:
Do you have any reason you ended up to list PodIPPools instead of selecting them by labels?

Reply:
@YutaroHayakawa thanks for the additional feedback. I can see how providing a list of pools can be a suboptimal UX when managing a large number of pools. On the other hand, the current implementation is more efficient. Is it worth providing both options or do you think the UX of label-based selection outweighs the efficiency of the current approach? Here's an example of supporting both selection types:
```go
type NodeSpec struct {
...
	// PodIPPools advertises CIDRS from CilimPodIPPools that have been allocated to the
	// CiliumNodes matching this policy.
	PodIPPools *PodIPPools
}

type PodIPPools struct {
	// NameSelector advertises CilimPodIPPools based on pools names. Ignored if LabelSelector is specified
	NameSelector []string

	// LabelSelector advertises CiliumPodIPPools based on labels....
	LabelSelector *slimv1.LabelSelector
}
```

We could initially omit `NameSelector` but still use `PodIPPools` to support other selection types in the future. Thoughts?

Reply:
Sounds nice, but sorry I changed my mind and thought your current implementation is better. Thinking well about the label-based design, I realized that it requires all agents to watch `CiliumPodIPPool`, which is not the case right now (currently, only Cilium Operator watches `CiliumPodIPPool`). This adds significant churn to API servers, especially in the large cluster, so I think the benefit exceeds the cost.

Reply:
@YutaroHayakawa 
>  I realized that it requires all agents to watch CiliumPodIPPool ... . This adds significant churn to API servers,

Is it really an issue? Each agent is already connected to the apiserver, and this just adds one more type to watch. I would not expect many updates happening on the `CiliumPodIPPool` CRs, IMO usually users set it one and would not touch that it even for months.

I just realized a similar issue could be then with my PR as well which now needs agents to watch for `CiliumLoadBalancerIPPool`: https://github.com/cilium/cilium/pull/27705

Reply:
> We could extend [IPAMPoolAllocation](https://github.com/cilium/cilium/blob/main/pkg/ipam/types/types.go#L59-L71) to include CiliumPodIPPool labels.

Yeah, if we want to avoid watching `CiliumPodIPPool`, that would be the way.

> Is it really an issue? Each agent is already connected to the apiserver, and this just adds one more type to watch. I would not expect many updates happening on the CiliumPodIPPool CRs, IMO usually users set it once and would not touch that it even for months.

Umm, I think you're right here. In the case of `CiliumPodIPPool`'s use case, as you said, it might not be a big issue. Since I don't fully understand the use cases of `CiliumPodIPPool` I'd like to hear @gandro 's opinion here.

Reply:
> We could extend [IPAMPoolAllocation](https://github.com/cilium/cilium/blob/main/pkg/ipam/types/types.go#L59-L71) to include CiliumPodIPPool labels.

[Here](https://github.com/danehans/cilium/pull/new/issue_26937_labels) is a WIP branch for implementing the label-based approach. I'm having issues updating the CiliumNode IPAM allocation pool labels based on CiliumPodIPPool labels but I think the implementation is close.

Reply:
> We could extend IPAMPoolAllocation to include CiliumPodIPPool labels. The IPAM allocator would then write the labels from CiliumPodIPPool to the CiliumNode.

I worry a bit that this will cause synchronization bugs, because we're now duplicating mutable information across CRDs. If we do this, it means we have to update all CiliumNode objects when a CiliumPodIPPool label changes, which is about as costly as just watching all CiliumPodIPPool from each agent. So I'm not sure what we will save by doing this.

It's too early to know all the different use cases of CiliumPodIPPools, but I expect there not to be a more than a few dozen per cluster. So I don't expect this to be too costly overall.